### DISCUSSION TITLE
Why is save() recommended over update methods?

### DISCUSSION THREAD

**Discussion by vresetnikov** [Automattic/mongoose#13221] [Q&A] [answered]
Hello,

The mongoose documentation states:

So mongoose says: 

_You should use save() rather than updateOne() and updateMany() where possible. However, Model.updateOne() and Model.updateMany() have a few advantages:

updateOne() is atomic. If you load a document using find(), it may change before you save() it.
updateOne() doesn't require you to load the document into memory, which may give you better performance if your documents are huge._

The docs don't explain why it is recommended to use `save()` instead of those other methods. Furthermore, the 2 advantages indicated seem to be sufficient to do the opposite - opt for those other methods as they are potentially safer and more performant. So what would be the main reasons for the `save()` recommendation?

**Comment by vkarpov15** [comment, marked as answer]
We'll add a little more info on that. Because you have the document in memory, `save()` gives you more complete validation (`updateOne()` can't validate `inc`, etc.) and better middleware support. So `save()` gives you access to more features, but `updateOne()` is faster and atomic.

**Comment by hembhagat99** [comment]
@vkarpov15 I think `updateOne` and `updateMany` should be recommended in the documentation since **_faster_** and **_atomic_** are requirements of almost all real-world applications. So, why recommend `save`?

Also, can you elaborate (or point me to documentation) why having validators on `updateOne` and `updateMany` is problematic?

**Reply by vkarpov15** [reply]
@hembhagat99 because `updateOne()` being atomic means that the document being updated isn't in Node.js, so Mongoose can't run validation logic on the document. 

" since faster and atomic are requirements of almost all real-world applications." <-- false. Fast enough is a requirement, and atomic is nice to have in most cases, especially if you turn on OCC. But up to you, if `save()` is too slow then you can always use `updateOne()`.
